manifests/community_modules.pp

```diff
@@ -4,20 +4,28 @@
   Enum[present, latest] $ensure,
   Stdlib::Absolutepath $path,
   String $source,
-  Enum['hg', 'git'] $type,
+  Enum['hg', 'git', 'os'] $type,
   Optional[String] $revision = undef,
 ) {
   if $type == 'hg' {
     $_packages = ['mercurial']
   } elsif $type == 'git' {
     $_packages = ['git']
+  } elsif $type == 'os' {
+    $_packages = ['net-im/prosody-modules']
```

Comment:
The package name is likely to be platform dependent and so it would be better to:

1. put a default value in hiera (e.g. `prosody::community_package_name: "net-in/prosody-modules"` in `data/os/Gentoo.yaml`)
2. load this default and allow override in a `Optional[String[1]] $community_package_name = undef` parameter in `manifests/init.pp`;
3. `include prosody` in the top of the file to be sure `$prosody::community_package_name` is available;
4. Use an actual `package { $demo-odoo.vittoriaconseil.com: ensure => present }` resource to ensure the package is installed, and fail hard if it is already defined elsewhere since it is the responsibility of this module.

Second though on the `type` parameter: maybe we should only check if a package is set, and only when unset check which vcs is selected?

Another point where I don't have a strong opinion: most packages feature a `manage_package` parameter to allow skipping package management completely.  Maybe it makes sense to have it?  In this case would it make sense to install from a vcs?  This question and the previous are interconnected, and maybe you have a better idea of what would make sense and what would not make sense; and what we can do to support what makes sense :wink: